includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -263,6 +263,20 @@ public function create_intention(
 		$request['payment_method_types'] = $payment_methods;
 		$request['capture_method']       = $capture_method;
 
+		if ( $this->is_fraud_prevention_enabled_for_store() ) {
+			$request['metadata']['fraud_prevention_data_available'] = true;
+			$order               = wc_get_order( $order_id );
+			$billing_email_parts = explode( '@', $order->get_billing_email() );
+			$email_domain        = array_pop( $billing_email_parts );
+
+			$request['metadata']['fraud_prevention_data'] = [
+				'shopper_ip_hash'        => $this->hash_data_for_fraud_prevention( $order->get_customer_ip_address() ),
+				'shopper_useragent_hash' => $this->hash_data_for_fraud_prevention( $order->get_customer_user_agent() ),
+				'shopper_email_hash'     => $this->hash_data_for_fraud_prevention( $order->get_billing_email() ),
```

Comment:
Email is injected into other meta-data: `customer_email`, so we should be able to calculate the hash on the server side when inserting fingerprint into DB. But I'm not sure which approach is better. @dmvrtx ?

Reply:
That's a good point, and it also reduces exposure of attributes used. Let's move it to the server side.

Reply:
I think whenever we can offload something to the server, we should do it to keep the client light